invokeai/app/invocations/ip_adapter.py

```diff
@@ -48,20 +49,25 @@ class IPAdapterOutput(BaseInvocationOutput):
     ip_adapter: IPAdapterField = OutputField(description=FieldDescriptions.ip_adapter, title="IP-Adapter")
 
 
+CLIP_VISION_MODEL_MAP = {"ViT-H": "ip_adapter_sd_image_encoder", "ViT-G": "ip_adapter_sdxl_image_encoder"}
+
+
 @invocation("ip_adapter", title="IP-Adapter", tags=["ip_adapter", "control"], category="ip_adapter", version="1.2.2")
 class IPAdapterInvocation(BaseInvocation):
     """Collects IP-Adapter info to pass to other nodes."""
 
     # Inputs
-    image: Union[ImageField, List[ImageField]] = InputField(description="The IP-Adapter image prompt(s).")
+    image: Union[ImageField, List[ImageField]] = InputField(description="The IP-Adapter image prompt(s).", ui_order=1)
     ip_adapter_model: ModelIdentifierField = InputField(
         description="The IP-Adapter model.",
         title="IP-Adapter Model",
         input=Input.Direct,
         ui_order=-1,
         ui_type=UIType.IPAdapterModel,
     )
-
+    clip_vision_model: Literal["ViT-H", "ViT-G"] = InputField(
+        description="CLIP Vision model to use", default="ViT-H", ui_order=2
```

Comment:
The description should probably explain that the `clip_vision_model` will be overridden if the right choice is known for the selected `ip_adapter_model`

Reply:
The issue here is that the setting is currently mandatory for checkpoint files but not required for the InvokeAI format. Maybe I make it so that the setting is used if the user explicitly sets it - no matter the format but if unset then the former takes over?

Reply:
I think it's fine as-implemented - just needs a little more explanation in the `description` field.

I suspect that it is possible to figure out the CLIP model, given that choosing the wrong one fails. But, it might be hard. Let's leave that as a future improvement.